.github/workflows/test.yaml

```diff
@@ -11,6 +11,15 @@ on:
     branches: [ trunk ]
 
 
+# Only have a run a single parallel for each branch.
+# Runs for trunk are queues.
+# Older runs for non-trunk branches are cancelled and the jobs are executed
+# only for the latest push to the branch.
+concurrency:
+  group: ${{ github.head_ref }}
+  cancel-in-progress: ${{ github.head_ref != 'refs/heads/trunk' }}
```

Comment:
It looks like `head_ref` gets set to `""` for...  manually...  re-run (?) trunk builds?  Just happened to be the first one I looked at.

https://github.com/twisted/twisted/runs/3274487611?check_suite_focus=true#step:7:152
```
      "head_ref": "",
```

Likewise though for what appears to be a merge-triggered build.

https://github.com/twisted/twisted/runs/3270940492?check_suite_focus=true#step:7:152
```
      "head_ref": "",
```

I think this check is wrong?

Reply:
Yes. The check is wrong. I think that `github.ref` is what we need :)
I have pushed an update.